### PR TITLE
[Accel] Add cohort dashboad manager

### PR DESCRIPTION
**Pull request by flying-pi**
**Description:** Added dashboard for cohort management
**Youtrack:** https://youtrack.raccoongang.com/issue/ACCEL-484

### CODE REVIEW

File: lms/djangoapps/cohort_management_dashboard/static/cohort_management_dashboard/css/index.css [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+.row {
+    border-right: 1px solid #c8c8c8;
+}
+
+.row_number_0 {
+        background-color: rgba(111, 183, 247, 0.33);
+}
+.row_number_1 {
+        background-color: rgba(145, 169, 214, 0.33);
+}
```
Comment (idegtiarov) [inline lms/djangoapps/cohort_management_dashboard/static/cohort_management_dashboard/css/index.css]: really?

File: lms/djangoapps/cohort_management_dashboard/static/cohort_management_dashboard/js/index.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                cohortID = options.cohort_id;
+                if (cohortID === undefined || cohortID === null || cohortID === 'None') {
+                    cohortID = cohortsList.cohorts.filter(emptyCohortFilter)[0].id;
+                }
+                bindCohortChangeLoster();
+                $slectControl.find('option[value="' + cohortID + '"]').prop('selected', true);
+                onCohortChanged(cohortID);
+            }
+
+            loadCohortsList().then(displaySelectControl);
```
Comment (idegtiarov) [inline lms/djangoapps/cohort_management_dashboard/static/cohort_management_dashboard/js/index.js]: Promise in not looks like required object here, is it correct

File: lms/djangoapps/cohort_management_dashboard/static/cohort_management_dashboard/templates/student_table_body.underscore [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+                    </option>
+                <% } %>
+            </select>
+            <input type="button" value="<%- gettext('Ok')%>" class="btn btn-primary change-cohort-btn" data-username="<%- users[i].username %>">
+          </td>
+          <td>
+                <div class="remove-user" data-username="<%- users[i].username %>" ><span class="icon fa fa-times-circle" aria-hidden="true"></span> <%- gettext('Delete') %></div>
+          </td>
+      </tr>
+  <% } %>
```
Comment (idegtiarov) [inline lms/djangoapps/cohort_management_dashboard/static/cohort_management_dashboard/templates/student_table_body.underscore]: oh! face palm! again

File: lms/djangoapps/cohort_management_dashboard/static/cohort_management_dashboard/templates/student_table_body.underscore [hunk 4]
```diff
@@ -0,0 +1,20 @@
+  <% for ( var i = 0; i < users.length; i++ ) { %>
+      <tr class="row row_number_<%- i%2 %>">
+        <td><%- users[i].username %></td>
+          <td>
+
+            <select class="action-cohort-select" name="select">
+                <% for ( var j = 0; j < cohorts.length; j++ ) { %>
+                    <option
+                        value="<%- cohorts[j].id %>" <%- (cohorts[j].id == currentCohortID)? 'selected':'' %> >
```
Comment (idegtiarov) [inline lms/djangoapps/cohort_management_dashboard/static/cohort_management_dashboard/templates/student_table_body.underscore]: I would really appreciate if we can add some spaces between the ternary operators, thank you in advance.

File: lms/djangoapps/cohort_management_dashboard/templates/cohort_management_dashboard/tab_index_page.html [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    </div>
+</div>
+
+
+<%block name="js_extra">
+
+
+<%static:require_module module_name="cohort_management_dashboard/js/index" class_name="CohortManagement">
+new CohortManagement({
+courseID: '${unicode(course.id) | n, js_escaped_string}',
```
Comment (idegtiarov) [inline lms/djangoapps/cohort_management_dashboard/templates/cohort_management_dashboard/tab_index_page.html]: Indentations and spaces are really welcome through the following block.

File: lms/djangoapps/cohort_management_dashboard/urls.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+from django.conf.urls import url
+
+from openedx.core.djangoapps.course_groups.views import users_in_cohort
+from . import views
+
+app_name = u'lms.djangoapps.cohort_management_dashboard'
+urlpatterns = [
+    url(r'^dashboard/(?P<cohort_id>\d+)$', views.DashboardIndex.as_view(), name='dashboard_with_cohort'),
+    url(r'^dashboard', views.DashboardIndex.as_view(), {'cohort_id': None}, name='dashboard'),
+    # Note(yura.braiko@raccongang.com) for some reason rout for all_user_list in edx-platform/lms/urls.py doesn't work.
```
Comment (idegtiarov) [inline lms/djangoapps/cohort_management_dashboard/urls.py]: Are we going to know the reason?

Comment (flying-pi) [reply lms/djangoapps/cohort_management_dashboard/urls.py]: Yes. Cohort's URL contains 2 patterns, which match to the same expression and as a result users_in_cohort become unreachable.

File: lms/djangoapps/cohort_management_dashboard/static/cohort_management_dashboard/js/index.js (hunk 2 above)
Comment (flying-pi) [reply lms/djangoapps/cohort_management_dashboard/static/cohort_management_dashboard/js/index.js]: Refactored

File: lms/djangoapps/cohort_management_dashboard/static/cohort_management_dashboard/templates/student_table_body.underscore (hunk 4 above)
Comment (flying-pi) [reply lms/djangoapps/cohort_management_dashboard/static/cohort_management_dashboard/templates/student_table_body.underscore]: fixed

**Pull request merged**